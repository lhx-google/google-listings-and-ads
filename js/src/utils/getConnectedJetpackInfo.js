/**
 * External dependencies
 */
import { __ } from '@wordpress/i18n';

/**
 * @typedef {import('.~/data/selectors').JetpackAccount} JetpackAccount
 */
/**
 * Get a connected jetpack information for display on UI.
 * This function should always be called after the jetpack is connected.
 *
 * @param {JetpackAccount} jetpack Connected jetpack account data.
 * @return {string} Return email if current user is jetpack owner.
 *                  Otherwise, return a successfully connected info only.
 */
export default function getConnectedJetpackInfo( jetpack ) {
	// Error-proofing for calling with not yet connect jetpack.
	if ( jetpack.active !== 'yes' ) {
		return '';
	}

	if ( jetpack.owner === 'yes' ) {
		return jetpack.email;
	}

	// Show a connected text instead of owner email to non jetpack owner users.
	return __(
		'Successfully connected through Jetpack',
		'google-listings-and-ads'
	);
}
