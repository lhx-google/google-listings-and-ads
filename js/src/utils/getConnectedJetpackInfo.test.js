/**
 * Internal dependencies
 */
import getConnectedJetpackInfo from './getConnectedJetpackInfo';

describe( 'getConnectedJetpackInfo', () => {
	it( 'When jetpack in not connected, should return an empty string', () => {
		expect( getConnectedJetpackInfo( { active: 'no' } ) ).toBe( '' );
	} );

	describe( 'For connected jetpack', () => {
		it( 'When current user is the jetpack owner, should return email', () => {
			const jetpack = {
				active: 'yes',
				owner: 'yes',
				email: 'i_am_jetpack_owner@example.com',
			};

			expect( getConnectedJetpackInfo( jetpack ) ).toBe( jetpack.email );
		} );

		it( 'When current user is not the jetpack owner, should return a successfully connected info', () => {
			const jetpack = {
				active: 'yes',
				owner: 'no',
				email: 'i_am_another_admin_user@example.com',
			};

			const info = getConnectedJetpackInfo( jetpack );

			expect( info ).not.toBe( jetpack.email );
			expect( info ).toMatchSnapshot();
		} );
	} );
} );
