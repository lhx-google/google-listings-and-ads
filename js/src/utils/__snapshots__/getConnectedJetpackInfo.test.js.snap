// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getConnectedJetpackInfo For connected jetpack When current user is not the jetpack owner, should return a successfully connected info 1`] = `"Successfully connected through Jetpack"`;
